Kubernetes/HOWTO-on-prem.md

```diff
@@ -102,91 +75,219 @@ sudo cp -r !(*.tgz) ${CNI_BIN}
 ls ${CNI_BIN}
 ```
 
-## Prepare the Master ##
-Copy the entire directory from [this repository](https://github.com/Microsoft/SDN/tree/master/Kubernetes/linux), which contains a lot of the setup scripts and Kubernetes files that are essential to this process. Check them all out to `~/kube/` and make the scripts executable. This entire directory will be getting mounted for a lot of the docker containers in future steps, so keep its structure the same as outlined in the guide.
-
-We'll be creating the service cluster on the virtual subnet 11.0/16. This isn't configurable, so if you want to tweak this, it'll require manual modification.
 
 ### Certificates ###
-First, prepare the certificates that will be used for nodes to communicate in the cluster. Run a the generate-certs.sh script, passing **your** external IP as its (only) parameter -- this is likely whatever the IP address of your `eth0` interface is. You can acquire this through `ifconfig` or through:
+First, acquire your local IP address, either via `ifconfig` or:
 
     $ ip addr show dev eth0
 
-if you already know the interface name. Then:
+if you already know the interface name. Then, set it as an environmental variable, `MASTER_IP`, because we'll be referencing it a lot in this guide. Note that you will need to set this again if you end your session. Now, prepare the certificates that will be used for nodes to communicate in the cluster:
 
+    ~/kube $ MASTER_IP=10.123.45.67   # example! replace
     ~/kube $ cd certs
-    ~/kube/certs $ ./generate-certs.sh 10.123.45.67   # example! replace
-
-### Allow Routing ###
-_This step may be optional, depending on whether or not you've configured your intended cluster subnet to be routable already_.
+    ~/kube/certs $ ./generate-certs.sh $MASTER_IP
 
-Windows nodes will each grab a /24 from the cluster CIDR, so it's 192.168.0.0/16, the first Windows node will use 192.168.0.1/24 for its pods. Pass the first two octets of your clister CIDR to generate the routes:
-
-    ~/kube $ generate-routes.sh 192.168
 
 ### Prepare Manifests & Addons ###
 In the `manifest` folder, run the Python script, passing your master IP and the _full_ cluster CIDR:
 
-    $ python2 generate.py 10.123.45.67 192.168.0.0/16
+    $ ./generate.py $MASTER_IP 192.168.0.0/16

```

Comment:
We need an extra option on this script to specify topology / configuration being used:
1) Upstream Routed (Static Routes in ToR)
2) Host-Gateway (Static Routes on each Host)
3) (Optional) Flannel with win-overlay CNI plugin

This will not work if user is running in "Upstream Routed" mode where static routes are configured in the ToR. In this case, kube-control-manager.yaml file should not have "--allocate-node-cidrs=True" or "--cluster-cidr=10.10.0.0/16". Instead, each instance of kubelet running on a particular node will need to have an option "--cidr=X" to specify the static CIDR

Reply:
Also, this version still specifies v1.7.3 for kube-apiserver instead of v1.9.0

Reply:
Yep, absolutely true. I added [this section](https://github.com/Microsoft/SDN/blob/6d1b667aa617e3371a67af30e12dd5f7a309517d/Kubernetes/HOWTO-on-prem.md#optional-customize-cluster-cidr) to do that manually (though I forgot to remove `--cluster-cidr`). 

You're saying we should make it parameterized instead, yes?

Reply:
Waiting on @nwoodmsft for the versioning. :)

Reply:
Ah... didn't see the next section below. Can we integrate that option into the script?

Reply:
Definitely. Since the script is getting a little more complicated, I'll take the time to make it a bit nicer.